### PR TITLE
Integrate AttributeFilterHandler with AttributeFilter

### PR DESCRIPTION
**Pull request by kandl**
- Complete integration of the AttributeFilterHandler in AttributeFilter
- Improve folder structure and naming of the attribute filter components
- InvertableList - cleanup, refactoring and more precise naming (InverableSelect)
- Collocate components props with default components
- useAttributeFilterHandler hook - initialization improvements
- Add useAttributeFilterController hook - for our and custom filter implementations
- Add AsyncList component
- Minor fixes

JIRA: RAIL-4255

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9157/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10988/ : FAILURE in 3m 50s

File: libs/sdk-ui-filters/src/AttributeFilter@next/AttributeFilterBase.tsx [hunk 1]
```diff
@@ -1,82 +1,304 @@
...
+    /**
+     * Customize attribute filter with a callback function to trigger when an error occurs while
+     * loading attribute elements.
+     */
+    onError?: (error: any) => void;
+
+    /**
+     * Customize attribute filter with a component to be rendered if attribute elements loading fails
+     */
+    ErrorComponent?: React.ComponentType<IAttributeFilterErrorProps>;
```
Comment (ivanmjartan) [inline libs/sdk-ui-filters/src/AttributeFilter@next/AttributeFilterBase.tsx:125]: I will split IAttributeFilterBaseProps interface into two. 

One interface could IAttributeFilterBaseCustomization and its content could be list customisable components and final  IAttributeFilterBase could be composition of this two interfaces

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeFilterConfigurationButton.tsx [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
-import { IAttributeFilterConfigurationButtonProps } from "./types";
+
+/**
+ * @internal
+ */
+export interface IAttributeFilterConfigurationButtonProps {
+    onConfiguration: () => void;
+}
 
 //TODO this component is prepared for Dashboard edit mode customization and maybe should be moved to Dashboard
```
Comment (ivanmjartan) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeFilterConfigurationButton.tsx:11]: remove this TODO

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeFilterDeleteButton.tsx [hunk 3]
```diff
@@ -2,10 +2,16 @@
...
 const ALIGN_POINTS = [{ align: "bc tc", offset: { x: -1, y: 5 } }];
 
+/**
+ * @internal
+ */
+export interface IAttributeFilterDeleteButtonProps {
+    onDelete: () => void;
+}
+
 //TODO this component is prepared for Dashboard edit mode customization and maybe should be moved to Dashboard
```
Comment (ivanmjartan) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeFilterDeleteButton.tsx:15]: remove this TODO

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9165/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10996/ : FAILURE in 5m 06s

File: libs/sdk-ui-kit/styles/scss/form.scss [hunk 4]
```diff
@@ -345,13 +345,13 @@ textarea {
...
             border: 1px solid $gd-input-text-hover-border;
-            transition: all 0.2s;
+            // transition: all 0.2s; TODO: disable transition only during the scroll
             background-color: $gd-color-text-light;
         }
 
         &::after {
             transform: scale(0);
-            transition: transform 0.2s;
+            // transition: transform 0.2s; TODO: disable transition only during the scroll
```
Comment (ivanmjartan) [inline libs/sdk-ui-kit/styles/scss/form.scss:354]: Is this change save? It looks like all forms input-radio-label,
.input-checkbox-label are now without transition

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9174/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11004/ : FAILURE in 13m 00s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9175/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11005/ : FAILURE in 10m 06s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9176/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11006/ : FAILURE in 19m 54s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9178/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11009/ : FAILURE in 21m 04s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9188/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11015/ : FAILURE in 3m 18s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9194/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11020/ : FAILURE in 29m 41s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9200/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11027/ : FAILURE in 23m 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9206/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11029/ : SUCCESS in 27m 49s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2828

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2609/ : SUCCESS in 3m 46s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2908/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1741/ : SUCCESS in 18m 45s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11030/ : SUCCESS in 21m 37s

**Pull request merged**